### ISSUE TITLE
Tests fail for Python 3.8, pypy and pypy3

### ISSUE THREAD

**Issue by jugmac00** [zopefoundation/zopetoolkit#37]
see
https://travis-ci.org/zopefoundation/zopetoolkit

**Comment by mgedmin** [comment]
Python 3.8 fails in zope.contenttype:

```
Failure in test test_add_one_file (zope.contenttype.tests.testContentTypes.ContentTypesTestCase)
Traceback (most recent call last):
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.contenttype-4.4-py3.8.egg/zope/contenttype/tests/testContentTypes.py", line 73, in test_add_one_file
    self._check_types_count(1)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.contenttype-4.4-py3.8.egg/zope/contenttype/tests/testContentTypes.py", line 33, in _check_types_count
    self.assertEqual(len(mimetypes.types_map),
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 905, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: 571 != 572
Failure in test test_add_two_files (zope.contenttype.tests.testContentTypes.ContentTypesTestCase)
Traceback (most recent call last):
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.contenttype-4.4-py3.8.egg/zope/contenttype/tests/testContentTypes.py", line 87, in test_add_two_files
    self._check_types_count(2)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.contenttype-4.4-py3.8.egg/zope/contenttype/tests/testContentTypes.py", line 33, in _check_types_count
    self.assertEqual(len(mimetypes.types_map),
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/python/3.8-dev/lib/python3.8/unittest/case.py", line 905, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: 572 != 574
```

pypy fails in zope.pagetemplate

```
/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.untrustedpython-4.0.0-py2.7.egg/zope/untrustedpython/rcompile.py:19: DeprecationWarning: The compiler package is deprecated and removed in Python 3.x.
  import compiler.pycodegen
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
Failure in test ZopeEngine (zope.pagetemplate.engine)
Failed doctest test for zope.pagetemplate.engine.ZopeEngine
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.pagetemplate-4.4.1-py2.7.egg/zope/pagetemplate/engine.py", line 308, in ZopeEngine
----------------------------------------------------------------------
File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.pagetemplate-4.4.1-py2.7.egg/zope/pagetemplate/engine.py", line 345, in zope.pagetemplate.engine.ZopeEngine
Failed example:
    str(type(r).__name__) in (
      ('_Proxy',) if HAVE_UNTRUSTED else
      ('builtin_function_or_method', 'method'))
Expected:
    True
Got:
    False
```

pypy3 fails in zope.security:

```
Error in test test_iteration_of_odict_items_keys_values (zope.security.tests.test_checker.TestChecker)
Traceback (most recent call last):
  File "/opt/python/pypy3.6-7.1.1/lib-python/3/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/opt/python/pypy3.6-7.1.1/lib-python/3/unittest/case.py", line 601, in run
    testMethod()
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/tests/test_checker.py", line 469, in test_iteration_of_odict_items_keys_values
    self._check_iteration_of_dict_like(odict)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/tests/test_checker.py", line 442, in _check_iteration_of_dict_like
    self.assertEqual([], list(proxy.items()))
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/proxy.py", line 32, in _wrapper
    checker.check(wrapped, name)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/checker.py", line 268, in check
    raise ForbiddenAttribute(name, object)
   - class: builtins.dict_itemiterator
   - type: builtins.dict_itemiterator
zope.security.interfaces.ForbiddenAttribute: ('__next__', <dict_itemiterator object at 0x0000000003f63300>)
Error in test test_iteration_of_odict_items_keys_values (zope.security.tests.test_checker.TestCheckerPy)
Traceback (most recent call last):
  File "/opt/python/pypy3.6-7.1.1/lib-python/3/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/opt/python/pypy3.6-7.1.1/lib-python/3/unittest/case.py", line 601, in run
    testMethod()
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/tests/test_checker.py", line 469, in test_iteration_of_odict_items_keys_values
    self._check_iteration_of_dict_like(odict)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/tests/test_checker.py", line 442, in _check_iteration_of_dict_like
    self.assertEqual([], list(proxy.items()))
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/proxy.py", line 32, in _wrapper
    checker.check(wrapped, name)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/checker.py", line 268, in check
    raise ForbiddenAttribute(name, object)
   - class: builtins.dict_itemiterator
   - type: builtins.dict_itemiterator
zope.security.interfaces.ForbiddenAttribute: ('__next__', <dict_itemiterator object at 0x00007fe3abb9b740>)
Error in test test_iteration_of_odict_items_keys_values (zope.security.tests.test_checker.TestWatchingChecker)
Traceback (most recent call last):
  File "/opt/python/pypy3.6-7.1.1/lib-python/3/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/opt/python/pypy3.6-7.1.1/lib-python/3/unittest/case.py", line 601, in run
    testMethod()
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/tests/test_checker.py", line 469, in test_iteration_of_odict_items_keys_values
    self._check_iteration_of_dict_like(odict)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/tests/test_checker.py", line 442, in _check_iteration_of_dict_like
    self.assertEqual([], list(proxy.items()))
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/proxy.py", line 32, in _wrapper
    checker.check(wrapped, name)
  File "/home/travis/build/zopefoundation/zopetoolkit/eggs/zope.security-4.3.1-py3.6.egg/zope/security/checker.py", line 268, in check
    raise ForbiddenAttribute(name, object)
   - class: builtins.dict_itemiterator
   - type: builtins.dict_itemiterator
zope.security.interfaces.ForbiddenAttribute: ('__next__', <dict_itemiterator object at 0x00007fe3abb00da0>)
```

**Comment by mgedmin** [comment]
zope.contenttype -> https://github.com/zopefoundation/zope.contenttype/issues/7

**Comment by cjwatson** [comment]
zope.pagetemplate → zopefoundation/zope.pagetemplate#22

zope.security was fixed in 5.0.

So I think we now just need releases of zope.contenttype and zope.pagetemplate, and to bump versions here.

**Comment by jugmac00** [comment]
Fixed by https://github.com/zopefoundation/zopetoolkit/pull/38 and others.

Thanks all!
